flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/dataset/DataSetSingleRowJoin.scala

```diff
@@ -46,6 +47,7 @@ class DataSetSingleRowJoin(
     rowRelDataType: RelDataType,
     joinCondition: RexNode,
     joinRowType: RelDataType,
+    joinType: JoinRelType,
```

Comment:
Good point, but maybe add additional check in other method? for example in `translateToPlan`. 
In this case if ``DataSetSingleRowJoin.computeSelfCost`` don't return best cost we don't get exception and don't select incorrect ``RelNode``. but if we whatever get best cost we can throw exception in  `translateToPlan` for incorrect joinType.
What do you mean about it?